### PR TITLE
mz: accept `psql` args in `shell` command

### PR DESCRIPTION
**Pull request by joacoc**
### Motivation

Accept passing `psql` args in the `shell` command.
https://github.com/MaterializeInc/materialize/issues/16987

**Usage**

Connect and interact with your Materialize database:

```bash
mz shell
```

Connect to a particular region:

```bash
mz shell --cloud-provider-region aws/us-east-1
```

Pass extra arguments to `psql`:

```bash
mz shell -l --csv
```

### Tips for reviewer

I had to turn `cloud_provider_region` as an `Option` command rather than an `<Optional> Arg`. 

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by mjibson — COMMENTED** [review]
Does this work with both present? `mz shell --cloud-provider-region aws/us-east-1 -l --csv`?

There's 5 changes to the clap arguments here, and it's hard to know how they interact. Do we need all 5 changes? Is `#[clap(multiple = true, last = true)]` sufficient?

File: src/mz/src/bin/mz/main.rs [hunk 1]
```diff
@@ -166,8 +167,14 @@ enum Commands {
     },
 
     /// Connect to a region using a SQL shell
+    #[clap(setting = AppSettings::TrailingVarArg, setting = AppSettings::AllowLeadingHyphen)]
     Shell {
+        // Select to a particular region
+        #[clap(long)]
```
[the remark this answers is not in the document]
Comment (joacoc) [reply src/mz/src/bin/mz/main.rs:173]: To avoid conflicts with the `-c/--command` flag from `psql`

**Comment by joacoc** [comment]
> Does this work with both present? mz shell --cloud-provider-region aws/us-east-1 -l --csv?

Works by adding a hyphen in-between:

`mz shell --cloud-provider-region aws/us-east -- -1 -l --csv`

I will revisit and retry the possible alternatives to reduce the derives in the args while keeping the feature.

**Comment by mjibson** [comment]
I'm very on board with forcing users to prefix these with `--` instead of have it only be sometimes needed. That would clearly split up mz args and psql args.

**Comment by joacoc** [comment]
Already implemented.

**Pull request closed without merging**